core/src/test/java/org/apache/iceberg/rest/requests/TestCreateNamespaceRequest.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.rest.requests;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.Map;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.util.JsonUtil;
+import org.assertj.core.api.Assertions;
+import org.junit.Test;
+
+public class TestCreateNamespaceRequest {
```

Comment:
just curious whether it would make sense to test more non-happy paths, where you're passing:
* a null/empty json
* a json with fields that that Ser/De components don't understand (think about somebody mistyping `namespace` or `properties`)
* a json where `namespace` / `properties` get strings that they can't interpret (such as numbers)

Reply:
* null/empty JSON - I agree we should make sure that required fields are present
* JSON with extra fields - I think extra fields should be ignored by the client so that we can evolve the payloads. Clients shouldn't fail if the service returns extra fields, and services shouldn't necessarily fail if the client sends extra fields.

Reply:
Added a few tests for malformed.

Specifically added:
- an empty json
- (String) null - simulating if the response entity has no body at all
-  fields are spelled incorrectly
- `namespace` / `properties` have fields that aren't the correct type.

Reply:
If we're good with the tests, I'll start adding them to more of the request / response objects.